atomicfu/src/commonMain/kotlin/kotlinx/atomicfu/Trace.kt

```diff
@@ -0,0 +1,62 @@
+package kotlinx.atomicfu
+
+import kotlin.js.JsName
+
+/**
+ * Creates `Trace` object for tracing atomic operations.
+ *
+ * ### Usage
+ *
+ * Create a separate field for `Trace`:
+ *
+ * ```
+ * val trace = Trace(size)
+ * ```
+ *
+ * Using it to add trace messages:
+ *
+ * ```
+ * trace { "Doing something" }
+ * ```
+ *
+ * Pass it to `atomic` constructor to automatically trace all modifications of the corresponding field:
+ *
+ * ```
+ * val a = atomic(initialValue, trace)
+ * ```
+ */
+@Suppress("FunctionName")
+@JsName("atomicfu\$Trace\$")
+public fun Trace(size: Int = 32, format: (Int, String) -> String = { index, text -> "$index: $text" }): Trace =
```

Comment:
This function must be `@InlineOnly` on JVM (otherwise its debug metadata pollutes the resulting class files and `AtomicfuBytecodeTest` would not pass)